packages/eslint-plugin/src/rules/strict-comparisons.ts

```diff
@@ -0,0 +1,310 @@
+     */
+    function getStrictestComparableType(
+      typesLeft: TypeKind[],
+      typesRight: TypeKind[],
+    ): TypeKind | undefined {
+      const overlappingTypes = typesLeft.filter(
+        type => typesRight.indexOf(type) >= 0,
+      );
+
+      if (overlappingTypes.length > 0) {
+        return getStrictestKind(overlappingTypes);
+      } else {
+        // In case one of the types is "any", get the strictest type of the other array
+        if (arrayContainsKind(typesLeft, TypeKind.Any)) {
+          return getStrictestKind(typesRight);
+        }
+        if (arrayContainsKind(typesRight, TypeKind.Any)) {
+          return getStrictestKind(typesLeft);
+        }
+
+        // In case one array contains NullOrUndefined and the other an Object, return Object
+        if (
+          (arrayContainsKind(typesLeft, TypeKind.NullOrUndefined) &&
+            arrayContainsKind(typesRight, TypeKind.Object)) ||
+          (arrayContainsKind(typesRight, TypeKind.NullOrUndefined) &&
+            arrayContainsKind(typesLeft, TypeKind.Object))
+        ) {
+          return TypeKind.Object;
+        }
+        return undefined;
+      }
+    }
+
+    /**
+     * Check if a specific TypeKind is present in an array
+     * @param types array of TypeKinds
+     * @param typeToCheck TypeKind to check
+     */
+    function arrayContainsKind(
+      types: TypeKind[],
+      typeToCheck: TypeKind,
+    ): boolean {
+      return types.some(type => type === typeToCheck);
+    }
+
+    /**
+     * Return the strictest kind of an array
+     * @param types array of TypeKinds
+     */
+    function getStrictestKind(types: TypeKind[]): TypeKind {
+      // tslint:disable-next-line:no-unsafe-any
+      return Math.max.apply(Math, types);
+    }
+
+    /**
+     * Check if the operator is a comparison operator
+     * @param operator the operator to check
+     */
+    function isComparisonOperator(operator: string): boolean {
+      if (isEqualityOperator(operator)) {
+        return true;
+      }
+      switch (operator) {
+        case '<':
+        case '>':
+        case '<=':
+        case '>=':
+          return true;
+        default:
+          return false;
+      }
+    }
+
+    /**
+     * Check if the operator is an equality operator
+     * @param operator the operator to check
+     */
+    function isEqualityOperator(operator: string): boolean {
+      switch (operator) {
+        case '==':
+        case '!=':
+        case '===':
+        case '!==':
+          return true;
+        default:
+          return false;
+      }
+    }
+
+    /**
+     * Helper function to get base type of node
+     * @param node the node to be evaluated.
+     */
+    function getNodeType(node: TSESTree.Node): ts.Type {
+      const tsNode = service.esTreeNodeToTSNodeMap.get(node);
+      return typeChecker.getTypeAtLocation(tsNode);
+    }
+
+    return {
+      BinaryExpression(node: TSESTree.BinaryExpression) {
```

Comment:
you can omit this type

Reply:
I don't think I can. If I change it to `BinaryExpression(node)` it tells me that `node` is of type `never`. Or am I missing something?

Reply:
I added those types myself and in my other PR I noticed that I was missing `BinaryExpression`

https://github.com/typescript-eslint/typescript-eslint/pull/315/files#diff-d69227db78f612ab8986f26cca8df9f4R410

Reply:
Ok, so I'll wait for your PR to be merged and update my branch. Any other feedback?

Reply:
Because this hasn't been merged yet, I guess we can wait and adjust it later?

Reply:
@AndreasGassmann - could you please duplicate the changes from #315 into here to fix this.
It removes the dependency on the other PR, and ensures we don't have to remember to do this later.